code/control/LDAPSecurityController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Class SAMLController
+ *
+ * This controller handles serving metadata requests for the IdP, as well as handling
+ * creating new users and logging them into SilverStripe after being authenticated at the IdP.
+ */
+class LDAPSecurityController extends Security {
+
+	/**
+	 * @var array
+	 */
+	private static $allowed_actions = array(
+		'index',
+		'lostpassword',
+		'LostPasswordForm',
+		'ChangePasswordForm'
+	);
+
+	/**
+	 * @param Member $member
+	 * @param $autologinToken
+	 * @return string
+	 */
+	public static function getPasswordResetLink($member, $autologinToken) {
+		$autologinToken = urldecode($autologinToken);
+		$selfControllerClass = __CLASS__;
+		$selfController = new $selfControllerClass();
+		return $selfController->Link('changepassword') . "?m={$member->ID}&t=$autologinToken";
+	}
+
+	/**
+	 * Factory method for the lost password form
+	 *
+	 * @return Form Returns the lost password form
+	 */
+	public function ChangePasswordForm() {
+		return Object::create('LDAPChangePasswordForm', $this, 'ChangePasswordForm');
+	}
+
+	/**
+	 * Factory method for the lost password form
+	 *
+	 * @return Form Returns the lost password form
+	 */
+	public function LostPasswordForm() {
+		$email = new EmailField('Email', _t('Member.EMAIL', 'Email'));
```

Comment:
Yeah, I'll work on it, in the forgot password box you'll supply a username instead, it will be looked up and synced, and an email will be sent to the user, if there is an email set up.